simple.js

```diff
@@ -496,6 +498,48 @@ define(function(require, exports, module) {
         }
 
         /**
+         * Detects Ace syntax mode from shebang for files with no extension
+         */
+        function detectSyntax() {
+            // syntax modes
+            var MODES = {
+                perl: /perl/,
+                php: /php/,
+                python: /python/,
+                ruby: /ruby/,
+                sh: /^(bash2?|ksh|sh)/
+            };
+
+            tabManager.on("open", function(e) {
+                var tab = e.tab;
+
+                // only handle files with no extensions that are open with Ace
+                if (tab.editorType !== "ace" || !_.isEmpty(extname(tab.path)))
+                    return;
+
+                var editor = tab.editor;
+                var session = editor.ace.getSession();
+
+                // ensure first line is a shebang
+                var firstLine = session.getLine(0);
+                if (firstLine.startsWith("#!")) {
+                    // turn lines like #!/usr/bin/env [VAR=val]... bash to #!/usr/bin/bash for easy matching
```

Comment:
This seems presumptuous if the binary doesn't actually live at that path. Does it not suffice to check for a format of

`#!/path/to/foo [bar ...]`

and then pattern match against `foo` if `bar` is `undefined`, else if `foo` is `env` and `bar` is not `undefined`, then pattern-match against `bar`?